packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -172,23 +237,31 @@ export function createTRPCOptionsProxy<TRouter extends AnyRouter>(
   return createRecursiveProxy(({ args, path: _path }) => {
     const path = [..._path];
     const utilName = path.pop() as UtilsMethods;
-    const [input, userOptions] = args as any[];
+    const [arg1, arg2] = args as any[];
 
     const queryType = getQueryType(utilName);
-    const queryKey = getQueryKeyInternal(path, input, queryType);
+    const queryKey = getQueryKeyInternal(path, arg1, queryType ?? 'any');
 
     const contextMap: Record<UtilsMethods, () => unknown> = {
+      '~input': undefined as any,
+      '~output': undefined as any,
+      queryKey: () => queryKey,
+      queryFilter: (): QueryFilters => {
+        return {
+          queryKey: queryKey,
+        };
+      },
       infiniteQueryOptions: () =>
```

Comment:
for consistency with others :)

```suggestion
      infiniteQueryOptions: () => {
        return trpcInfiniteQueryOptions({
          opts: arg2,
          path,
          queryClient: opts.queryClient,
          queryKey: queryKey,
          query: callIt('query'),
        });
      },
```